rpcserver.go

```diff
@@ -7136,8 +7141,19 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 
 	// Now that we have the accepted quote, we know the amount in Satoshi
 	// that we need to pay. We can now update the invoice with this amount.
-	mSatPerUnit := acceptedQuote.AskPrice
-	iReq.ValueMsat = int64(req.AssetAmount * mSatPerUnit)
+	askAssetRate := rfqmath.FixedPoint[rfqmath.GoInt[uint64]]{
+		Coefficient: rfqmath.NewGoInt(
+			acceptedQuote.AskAssetRate.Coefficient,
+		),
+		Scale: uint8(acceptedQuote.AskAssetRate.Scale),
+	}
+	assetAmount := rfqmath.FixedPoint[rfqmath.GoInt[uint64]]{
+		Coefficient: rfqmath.NewGoInt(req.AssetAmount),
+		Scale:       0,
```

Comment:
Same here again re assuming scale.

Reply:
Don't we have to assume a scale of 0 when converting an asset amount to a fixed-point?